src/GridFS/StreamWrapper.php

```diff
@@ -123,14 +148,71 @@ public function stream_eof(): bool
      */
     public function stream_open(string $path, string $mode, int $options, ?string &$openedPath): bool
     {
-        $this->initProtocol($path);
+        [$protocol] = explode('://', $path, 2);
+
+        $context = [];
+        if (is_resource($this->context)) {
+            $context = stream_context_get_options($this->context)[$protocol] ?? [];
+
+            if (! is_array($context)) {
+                if ($options & STREAM_REPORT_ERRORS) {
+                    trigger_error(sprintf('Invalid context for "%s" protocol.', $protocol), E_USER_ERROR);
```

Comment:
Let's use exceptions instead. This flag is never sent. https://jira.mongodb.org/browse/PHPLIB-1318